src/lib/application.ts

```diff
@@ -525,6 +525,14 @@ export class Application extends ChildableComponent<
         const start = Date.now();
         out = Path.resolve(out);
         await this.renderer.render(project, out);
+
+        // Generate a message summarizing the number of errors and warnings if there are any
```

Comment:
I think this is the wrong place to put this. TypeDoc has a `--watch` option, which will result in `generateDocs` and `generateJson` being called many times. If a user has both options turned on, then with this implementation they may get two warnings/errors for each compilation.

I think the way I'd prefer to handle this is to add a public `logRunSummary` (better name?) method on `Application` which is called in `cli.ts` in the appropriate places

Reply:
This also has the benefit of making the tests for this feature much less hacky - I've very intentionally avoided mocking pieces of TypeDoc in tests (almost) everywhere so far, they tend to lead to testing implementation details+easily broken tests

Reply:
> I think this is the wrong place to put this. TypeDoc has a `--watch` option, which will result in `generateDocs` and `generateJson` being called many times. If a user has both options turned on, then with this implementation they may get two warnings/errors for each compilation.
> 
> I think the way I'd prefer to handle this is to add a public `logRunSummary` (better name?) method on `Application` which is called in `cli.ts` in the appropriate places

That seems reasonable. I'll take a look at it.